kubernetes/resource_kubernetes_horizontal_pod_autoscaler.go

```diff
@@ -106,25 +115,35 @@ func resourceKubernetesHorizontalPodAutoscalerCreate(d *schema.ResourceData, met
 }
 
 func resourceKubernetesHorizontalPodAutoscalerRead(d *schema.ResourceData, meta interface{}) error {
+	if useV2(d) {
+		return resourceKubernetesHorizontalPodAutoscalerV2Read(d, meta)
+	}
+
 	conn := meta.(*KubeClientsets).MainClientset
 
 	namespace, name, err := idParts(d.Id())
 	if err != nil {
 		return err
 	}
 	log.Printf("[INFO] Reading horizontal pod autoscaler %s", name)
-	svc, err := conn.AutoscalingV1().HorizontalPodAutoscalers(namespace).Get(name, meta_v1.GetOptions{})
+	hpa, err := conn.AutoscalingV1().HorizontalPodAutoscalers(namespace).Get(name, meta_v1.GetOptions{})
 	if err != nil {
 		log.Printf("[DEBUG] Received error: %#v", err)
 		return err
 	}
-	log.Printf("[INFO] Received horizontal pod autoscaler: %#v", svc)
-	err = d.Set("metadata", flattenMetadata(svc.ObjectMeta, d))
+
+	// NOTE: this is needed for import
+	if _, exists := hpa.ObjectMeta.GetAnnotations()["autoscaling.alpha.kubernetes.io/metrics"]; exists {
```

Comment:
An interesting quirk of the HPA API is that you can do a GET for an HPA using either version of the API, and it will mutate the response to have this annotation for v1 and to use the metrics field for v2. So this is the only way i see when importing to check if an HPA needs v2.

I wasn't sure if this belongs here, or if we should create a custom importer and check for this there.